fdbserver/VersionedBTree.actor.cpp

```diff
@@ -7702,11 +7702,16 @@ class KeyValueStoreRedwood : public IKeyValueStore {
 		        : SERVER_KNOBS->REDWOOD_REMAP_CLEANUP_WINDOW_BYTES;
 
 		EncodingType encodingType = EncodingType::XXHash64;
-
-		// Deterministically enable encryption based on uid
-		if (g_network->isSimulated() && logID.hash() % 2 == 0) {
-			encodingType = EncodingType::XOREncryption;
-			m_keyProvider = std::make_shared<XOREncryptionKeyProvider>(filename);
+		// TODO(yiwu): We should enable encryption only when tenant is in required mode, but currently when existing
```

Comment:
May be we want to update the comment here to the following:

Encryption will work in the following ways:
1. Mutation belonging to a tenant, will leverage per-tenant encryption key for encryption.
2. Otherwise, default tenant encryption domain will be used (TODO, yet to be added to the code)

Reply:
Will update the comment to reflect the updated plan, but the comment is about the issue where on worker server restart when we reopen Redwood, we don't even know the tenant mode to encrypt/decrypt correctly.